engine/apps/api/views/labels.py

```diff
@@ -51,88 +51,95 @@ class LabelsViewSet(LabelsFeatureFlagViewSet):
     def get_keys(self, request):
         """List of labels keys"""
         organization = self.request.auth.organization
-        result, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_keys()
-        return Response(result, status=response.status_code)
+        keys, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_keys()
+        return Response(keys, status=response.status_code)
 
-    @extend_schema(responses=LabelKeyValuesSerializer)
+    @extend_schema(responses=LabelOptionSerializer)
     def get_key(self, request, key_id):
-        """Key with the list of values"""
+        """
+        get_key returns LabelOption – key with the list of values
+        """
         organization = self.request.auth.organization
-        result, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_values(key_id)
-        self._update_labels_cache(result)
-        return Response(result, status=response.status_code)
+        label_option, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_label_by_key_id(
+            key_id
+        )
+        self._update_labels_cache(label_option)
+        return Response(label_option, status=response.status_code)
 
     @extend_schema(responses=LabelValueSerializer)
     def get_value(self, request, key_id, value_id):
-        """Value name"""
+        """get_value returns a Value"""
         organization = self.request.auth.organization
-        result, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_value(key_id, value_id)
-        self._update_labels_cache(result)
-        return Response(result, status=response.status_code)
+        value, response = LabelsAPIClient(organization.grafana_url, organization.api_token).get_value(key_id, value_id)
+        # TODO: update_labels_cache expects LabelOption, but get value returns a Value. Investigate, temporary disable.
```

Comment:
Does this TODO need to be addressed before merging the PR?

Reply:
No, it's for future iterations.